src/components/pre-search/manage-pre-search.vue

```diff
@@ -0,0 +1,60 @@
+<template>
+  <div v-if="!$x.query.searchBox">
+    <h1 v-if="!hasQueryPreviews" class="x-title1 max-desktop:x-title1-sm max-desktop:x-px-16">
+      {{ $t('popularSearches.title') }}
+    </h1>
+    <CustomQueryPreview :queriesPreviewInfo="queriesPreviewToRender" />
+  </div>
+</template>
+
+<script lang="ts">
+  import { computed, defineComponent, inject } from 'vue';
+  import { QueryPreviewInfo } from '@empathyco/x-components/queries-preview';
+
+  import { PopularSearchesState } from '@empathyco/x-components/types';
+  import { XPlugin } from '@empathyco/x-components';
+  import { popularSearchesXModule } from '@empathyco/x-components/popular-searches';
+  import { useStore } from '../../composables/use-store.composable';
+  import CustomQueryPreview from './custom-query-preview.vue';
+
+  export default defineComponent({
+    components: {
+      CustomQueryPreview
+    },
+    props: {
+      maxPopularSearchesToRender: {
+        type: Number,
+        default: 4
+      }
+    },
+    setup(props) {
+      XPlugin.registerXModule(popularSearchesXModule);
+      const injectedQueriesPreviewInfo = computed<QueryPreviewInfo[]>(() => {
+        const injectedQueriesPreview = inject<QueryPreviewInfo[] | { value: QueryPreviewInfo[] }>(
+          'queriesPreviewInfo',
+          []
+        );
+        return 'value' in injectedQueriesPreview
+          ? injectedQueriesPreview.value
+          : injectedQueriesPreview;
+      });
+
+      const hasQueryPreviews = computed<boolean>(
+        () => injectedQueriesPreviewInfo.value.length !== 0
```

Comment:
If the user does not provide queryPreviews at all in the snippet this will fail and nothing will be shown. We should be ready for that, since it is precisely the use case we want. In the App.vue you can use an empty array as fallback for the provider.